bundle/definition/validation.go

```diff
@@ -0,0 +1,69 @@
+package definition
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/qri-io/jsonschema"
+)
+
+// Validate applies JSON Schema validation to the data passed as a paraemeter.
+// If validation errors occur, they will be returned in as an error.
+func (s *Schema) Validate(data interface{}) error {
+
+	b, err := json.Marshal(s)
+	if err != nil {
+		return errors.Wrap(err, "unable to load schema")
+	}
+	def := new(jsonschema.RootSchema)
+	err = json.Unmarshal([]byte(b), def)
+	if err != nil {
+		return errors.Wrap(err, "unable to build schema")
+	}
+	payload, err := json.Marshal(data)
+	if err != nil {
+		return errors.Wrap(err, "unable to process data")
+	}
+	valErrs, err := def.ValidateBytes(payload)
+	if err != nil {
+		return errors.Wrap(err, "unable to perform validation")
+	}
+	if len(valErrs) > 0 {
+		var anError error
+		for _, err := range valErrs {
+			if anError == nil {
+				anError = errors.New(fmt.Sprintf("unable to validate %s, error: %s", err.PropertyPath, err.Message))
+			} else {
+				anError = errors.Wrap(anError, fmt.Sprintf("unable to validate %s, error: %s", err.PropertyPath, err.Message))
+			}
+		}
+		return errors.Wrap(anError, "invalid parameter or output")
```

Comment:
This is going to be incredibly difficult for a person to figure out what's wrong with their bundle, and maybe impossible for a tool because it's not structured. Long term, do you think we could return a structured message that tells them exactly what is wrong with property so that they can examine it and unpack each property and validation error?

Reply:
Yeah, can change to that.

Reply:
Refactored this part. Now returns an slice of ValidationErrors{ Path string, Error string}